src/TestSuite/Migrator.php

```diff
@@ -0,0 +1,253 @@
+    {
+        $files = (array)$file;
+
+        $migrator = new static($verbose);
+        $schemaCleaner = new SchemaCleaner($migrator->io);
+        $schemaCleaner->dropTables($connectionName);
+
+        foreach ($files as $file) {
+            if (!file_exists($file)) {
+                throw new \RuntimeException('The file ' . $file . ' could not found.');
+            }
+
+            $sql = file_get_contents($file);
+            if ($sql === false) {
+                throw new \RuntimeException('The file ' . $file . ' could not read.');
+            }
+
+            ConnectionManager::get($connectionName)->execute($sql);
+
+            $migrator->io->success(
+                'Dump of schema in file ' . $file . ' for connection ' . $connectionName . ' successful.'
+            );
+        }
+
+        $schemaCleaner->truncateTables($connectionName);
+    }
+
+    /**
+     * Run migrations for all configured migrations.
+     *
+     * @param string[] $config Migration configuration.
+     * @return void
+     */
+    protected function runMigrations(array $config): void
+    {
+        $migrations = new Migrations();
+        $result = $migrations->migrate($config);
+
+        $msg = 'Migrations for ' . $this->stringifyConfig($config);
+
+
+        if ($result === true) {
+            $this->io->success($msg . ' successfully run.');
+        } else {
+            $this->io->error( $msg . ' failed.');
+        }
+    }
+
+    /**
+     * If a migration is missing or down, all tables of the considered connection are dropped.
+     *
+     * @return $this
+     * @throws \Exception
+     */
+    protected function handleMigrationsStatus(): self
+    {
+        $schemaCleaner = new SchemaCleaner($this->io);
+        $connectionsToDrop = [];
+        foreach ($this->getConfigs() as &$config) {
+            $connectionName = $config['connection'] = $config['connection'] ?? 'test';
+            $this->io->info("Reading migrations status for {$this->stringifyConfig($config)}...");
+            $migrations = new Migrations($config);
+            if ($this->isStatusChanged($migrations)) {
+                if (!in_array($connectionName, $connectionsToDrop))
+                {
+                    $connectionsToDrop[] = $connectionName;
+                }
+            }
+        }
+
+        if (empty($connectionsToDrop)) {
+            $this->io->success("No migration changes detected.");
+
+            return $this;
+        }
+
+        foreach ($connectionsToDrop as $connectionName) {
+            $schemaCleaner->dropTables($connectionName);
+        }
+
+        foreach ($this->getConfigs() as $config) {
+            $this->runMigrations($config);
+        }
+
+        // Truncate all created tables, except migration tables
+        foreach ($connectionsToDrop as $connectionName) {
+            $schema = ConnectionManager::get($connectionName)->getSchemaCollection();
```

Comment:
Should this create output letting the user know we've truncated tables?

Reply:
This will output information via the SchemaCleaner. So no need to add any additional output, if this was your concern.